packages/vue-app/template/client.js

```diff
@@ -599,42 +654,62 @@ function addHotReload($component, depth) {
       next: next.bind(this)
     })
     const context = app.context
+
     <% if (loading) { %>
-    if (this.$loading.start && !this.$loading.manual) this.$loading.start()
+    if (this.$loading.start && !this.$loading.manual) {
+      this.$loading.start()
+    }
     <% } %>
+
     callMiddleware.call(this, Components, context)
     .then(() => {
+      <% if (features.layouts) { %>
       // If layout changed
-      if (depth !== 0) return Promise.resolve()
+      if (depth !== 0) {
+        return true
+      }
+
       let layout = Component.options.layout || 'default'
       if (typeof layout === 'function') {
         layout = layout(context)
       }
-      if (this.layoutName === layout) return Promise.resolve()
+      if (this.layoutName === layout) {
+        return true
```

Comment:
Does `true` value means anything special? Before it was resolving to undefined.

Reply:
No, actually not. Have removed that